node/npm_dependencies.ts

```diff
@@ -14,6 +15,47 @@ import { Logger } from './logger.js'
 
 const TYPESCRIPT_EXTENSIONS = new Set(['.ts', '.cts', '.mts'])
 
+/**
+ * Returns the name of the `@types/` package used by a given specifier. Most of
+ * the times this is just the specifier itself, but scoped packages suffer a
+ * transformation (e.g. `@netlify/functions` -> `@types/netlify__functions`).
+ * https://github.com/DefinitelyTyped/DefinitelyTyped#what-about-scoped-packages
+ */
+const getTypesPackageName = (specifier: string) => {
+  if (!specifier.startsWith('@')) return path.join('@types', specifier)
+  const [scope, pkg] = specifier.split('/')
+  return path.join('@types', `${scope.replace('@', '')}__${pkg}`)
+}
+
+/**
+ * Starting from a `package.json` file, this tries detecting a TypeScript declaration file.
+ * It first looks at the `types` and `typings` fields in `package.json`.
+ * If it doesn't find them, it falls back to DefinitelyTyped packages (`@types/...`).
+ */
+const getTypesPath = async (packageJsonPath: string): Promise<string | undefined> => {
```

Comment:
This is totally a nit, since you shouldn't have to follow any formatting conventions that aren't enforced by our Prettier/ESLint configurations. Here goes anyway, in the spirit of sharing feedback.

I find this function a lot harder to parse than it should be, mostly because the code is super compact. Having some whitespace in between the different "sections" of the logic would help, as would things like separating the condition from the statement in `if` clauses.

Also, if you had the package.json parsing and the `@types/` parsing on two different functions, you could have much shorter variable names (e.g. `packageJsonPath` and `packageJsonContents` could simply be `path` and `contents`), which I think would also help.

Definitely not a blocker, though.

Reply:
Agree, that makes things a lot easier. Thanks for the feedback! I guess I wanted to get this mergeable, so I didn't look too much at readability. Also, it's been a while since I've written PRs with more than 100 lines of new code 😅

Reply:
Done in https://github.com/netlify/edge-bundler/pull/505/commits/aa604a8420a24f926db61f6ed56f02d6b61370f7